sled-agent/src/common/instance.rs

```diff
@@ -890,191 +603,109 @@ mod test {
     #[test]
     fn rude_terminate_of_migration_target_does_not_transition_instance() {
         let mut state = make_migration_target_instance();
-        assert_eq!(state.propolis_role(), PropolisRole::MigrationTarget);
 
         let prev = state.clone();
         state.terminate_rudely();
 
         assert_state_change_has_gen_change(&prev, &state);
-        assert_eq!(state.instance.gen, prev.instance.gen);
 
         // The migration state should transition.
         let migration =
-            state.migration.expect("instance must have a migration state");
+            state.migration_in.expect("instance must have a migration state");
         let prev_migration =
-            prev.migration.expect("previous state must have a migration");
+            prev.migration_in.expect("previous state must have a migration");
         assert_eq!(migration.state, MigrationState::Failed);
         assert!(migration.gen > prev_migration.gen);
     }
 
     #[test]
+    #[ignore = "this logic is basically trivial now, maybe just get rid of the test?"]
```

Comment:
I agree, this test isn't doing much lifting now--the main point of the test was to verify that a migration in followed by a migration out produced all the right instance runtime state transitions, which transitions now mostly live up in Nexus.